pkg/operator/controllers/dnsmasq/scripts/99-dnsmasq-restart.gotmpl

```diff
@@ -0,0 +1,45 @@
+{{ define "99-dnsmasq-restart" }}
+#!/bin/sh
+# This is a NetworkManager dispatcher script to restart dnsmasq
+# in the event of a network interface change (e. g. host servicing event https://learn.microsoft.com/en-us/azure/developer/intro/hosting-apps-on-azure)
+# this will restart dnsmasq, reapplying our /etc/resolv.conf file and overwriting any modifications made by NetworkManager
+
+interface=$1
+action=$2
+
+log() {
+    logger -i "$0" -t '99-DNSMASQ-RESTART SCRIPT' "$@"
+}
+
+# log dns configuration information relevant to SRE while troubleshooting
+# The line break used here is important for formatting
+check_dns_files() {
+    log "/etc/resolv.conf contents
+
+    $(cat /etc/resolv.conf)"
+
+    log "$(echo -n \"/etc/resolv.conf file metadata: \") $(ls -lZ /etc/resolv.conf)"
+
+    log "/etc/resolv.conf.dnsmasq contents
+
+    $(cat /etc/resolv.conf.dnsmasq)"
+
+    log "$(echo -n "/etc/resolv.conf.dnsmasq file metadata: ") $(ls -lZ /etc/resolv.conf.dnsmasq)"
+}
+
+if [[ $interface == eth* && $action == "up" ]] || [[ $interface == eth* && $action == "down" ]] || [[ $interface == enP* && $action == "up" ]] || [[ $interface == enP* && $action == "down" ]]; then
```

Comment:
It looks like we are proceeding from here no matter what (if the NIC is up, we proceed, if the NIC is down, we proceed), so do we need this `if` statement?

Reply:
There are a number of other things it can be, checkout the documentation for available actions: https://networkmanager.dev/docs/api/latest/NetworkManager-dispatcher.html